### PR TITLE
Enable multi-level nav in the left nav bar

### PR DESCRIPTION
**Pull request by svekars**
- Override pytorch_sphinx_theme to enable multi-level nav
- Add the needed CSS to indent the 2nd level to the right
- Use titlesonly in global_toc
- Turn introyt.rst to .py for the expansion work correctly.
How it will work:

https://github.com/user-attachments/assets/6b5391f8-0148-49d6-be50-551291651a2e

### CODE REVIEW

File: beginner_source/basics/intro.py [hunk 1]
```diff
@@ -13,9 +13,9 @@
 ===================
 
 Authors:
-`Suraj Subramanian <https://github.com/suraj813>`_,
+`Suraj Subramanian <https://github.com/subramen>`_,
 `Seth Juarez <https://github.com/sethjuarez/>`_,
-`Cassie Breviu <https://github.com/cassieview/>`_,
+`Cassie Breviu <https://github.com/cassiebreviu/>`_,
```
Comment (svekars) [inline beginner_source/basics/intro.py:18]: @cassiebreviu - can you confirm this is the new account that should be listed here.

Comment (cassiebreviu) [reply beginner_source/basics/intro.py:18]: Yes `https://github.com/cassiebreviu/` is correct. Thanks!

**Comment by svekars** [comment]
Linkcheck fails because the link doesn't yet exist. It will exist when this PR is merged

File: beginner_source/introyt/introyt.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+This tutorial follows along with the `PyTorch Beginner Series <https://www.youtube.com/playlist?list=PL_lsbAsL_o2CTlGHgMxNrKhzP97BaG9ZN>`_ on YouTube.
+
+`This tutorial assumes a basic familiarity with Python and Deep Learning concepts.`
+
+Running the Tutorial Code
+-------------------------
+You can run this tutorial in a couple of ways:
+
+- **In the cloud**: This is the easiest way to get started! Each section has a Colab link at the top, which opens a notebook with the code in a fully-hosted environment. Pro tip: Use Colab with a GPU runtime to speed up operations *Runtime > Change runtime type > GPU*
```
Comment (janeyx99) [inline beginner_source/introyt/introyt.py:24]:
```suggestion
- **On the cloud**: This is the easiest way to get started! Each section has a Colab link at the top, which opens a notebook with the code in a fully-hosted environment. Pro tip: Use Colab with a GPU runtime to speed up operations *Runtime > Change runtime type > GPU*
```

File: beginner_source/introyt/introyt.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- **Locally**: This option requires you to setup PyTorch and torchvision first on your local machine (`installation instructions <https://pytorch.org/get-started/locally/>`_). Download the notebook or copy the code into your favorite IDE.
```
Comment (janeyx99) [inline beginner_source/introyt/introyt.py:25]:
```suggestion
- **Locally**: This option requires you to set up PyTorch and torchvision on your local machine (`installation instructions <https://pytorch.org/get-started/locally/>`_). Download the notebook or copy the code into your favorite IDE.
```
nit

File: beginner_source/introyt.rst [hunk 4]
```diff
@@ -1,29 +1,10 @@
...
-`Brad Heintz <https://github.com/fbbradheintz>`_
-
-This tutorial follows along with the `PyTorch Beginner Series <https://www.youtube.com/playlist?list=PL_lsbAsL_o2CTlGHgMxNrKhzP97BaG9ZN>`_ on YouTube.
-
-`This tutorial assumes a basic familiarity with Python and Deep Learning concepts.`
-
-Running the Tutorial Code
--------------------------
-You can run this tutorial in a couple of ways:
+This page has been moved.
```
Comment (janeyx99) [inline beginner_source/introyt.rst:4]: Woah, would this autoredirect?

**Review by janeyx99 — COMMENTED** [review]
image

It'd be good if the concepts under "Learn the Basics" were automatically expanded, to give people an immediate summary of PyTorch basics just by looking at the sidebar. Or if these pages are as they were before + not nested under "Learn the Basics".

**Comment by janeyx99** [comment]
Let some review on the details--the multilevel nav bar is p cool. If we are only rearranging the sidebar, I think this is an improvement. There is still the greater task of deduplicating the content of the blitz, youtube, and qs basics, which I figure is outside the scope of the PR.

File: beginner_source/introyt.rst (hunk 4 above)
Comment (svekars) [reply beginner_source/introyt.rst:4]: Yes, tested in other pages like this.

**Comment by svekars** [comment]
> It'd be good if the concepts under "Learn the Basics" were automatically expanded, to give people an immediate summary of PyTorch basics just by looking at the sidebar. Or if these pages are as they were before + not nested under "Learn the Basics".

Because of how Sphinx works, it'd be tricky to implement.

**Comment by svekars** [comment]
> It'd be good if the concepts under "Learn the Basics" were automatically expanded, to give people an immediate summary of PyTorch basics just by looking at the sidebar. Or if these pages are as they were before + not nested under "Learn the Basics".

I can add a + sign like this:

Screenshot 2024-10-30 at 4 22 38 PM (2)

**Review by janeyx99 — APPROVED** [review]
preview looks alright to me

**Pull request merged**